Bug fixes and improvements

marcello3d opened the pull request:
parse coverage:
- return 100 when file is empty/completely skipped (was returning 1)
- 100% coverage for parse-coverage.js
- fix relative path calculation in github-comment.js
- compute common root

format coverage:
- new emojis in some cases
- fix path calculation for links to files in folders
- render main coverage link outside of pre tag
- don't show unchanged folders in overview
- add folder coverage under details
- 100% coverage for format-coverage.js
- fix links when common root has no coverage
- render markdown instead of html where possible

Commit:
parse-coverage fixes

- return 100 when file is empty/completely skipped (was returning 1)
- 100% coverage for parse-coverage.js
- fix relative path calculation in github-comment.js

Commit:
improve format-coverage

- new emojis in some cases
- fix path calculation for links to files in folders
- render main coverage link outside of pre tag
- don't show unchanged folders in overview
- add folder coverage under details
- 100% coverage for format-coverage.js

Commit:
standardjs + update snapshots

Commit:
add precommit checks for standard

viv-9000 commented:
<a>
  <strong><a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/26/artifacts/0//coverage/lcov-report/index.html">Code Coverage</a></strong> 
  from Circle CI <a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/26">build 26</a>
  
</p>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/26/artifacts/0//coverage/lcov-report/index.html">Project Coverage</a>    59.89% 💛 
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/26/artifacts/0//coverage/lcov-report/src/index.html">src/</a>    59.89% 💛 
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/26/artifacts/0//coverage/lcov-report/src/index.html">src/</a>    59.89% 💛 
</pre>
</details>

Commit:
tweak comment rendering

- don't show change stats if there is no prior coverage to compare to

Commit:
more logging/cleanup around artifact loading

Commit:
update snapshots

viv-9000 commented:
<a>
  <strong><a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/29/artifacts/0//coverage/lcov-report/index.html">Code Coverage</a></strong> 
  from Circle CI <a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/29">build 29</a>
  

  58.22% 💛 
<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/29/artifacts/0//coverage/lcov-report/src/index.html">src/</a>    58.22% 💛 
</pre>
</details>

Commit:
100% coverage on format-coverage.js again

Commit:
move folders around to have more interesting coveraeg

Commit:
fix links to find common root

- add dependency on istanbul-lib-report
- compute common root when parsing coverage file
- simplify link computation
- more tests
- render markdown instead of html where possible

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/32/artifacts/0//coverage/lcov-report/index.html)** 
  from Circle CI [build 32](https://circleci.com/gh/vivlabs/coverage-github-reporter/32)
  

**  58.74% 💛 **
<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/32/artifacts/0//coverage/lcov-report/index.html">src/         </a>     0.00% ❌ 
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/32/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/</a>   100.00% ✅ 
</pre>
</details>

Commit:
move more logic to github-comment file

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/33/artifacts/0//coverage/lcov-report/index.html): 57.84% 💛 ** 

<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/33/artifacts/0//coverage/lcov-report/index.html">src/         </a>     0.00% ❌ 
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/33/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/</a>   100.00% ✅ 
</pre>
</details>
From Circle CI [build 33](https://circleci.com/gh/vivlabs/coverage-github-reporter/33)
   — generated by [coverage-github-reporter](https://github.com/vivlabs/coverage-github-reporter)

Commit:
remove trailing spaces

Commit:
tweak github comment formatting + test coverage

Commit:
tweak formatting

Commit:
fix non-workflow artifact search

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/37/artifacts/0//coverage/lcov-report/index.html): 67.73% 💛** 

<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/37/artifacts/0//coverage/lcov-report/index.html">src/         </a>    23.40% 💔
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/37/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/</a>   100.00% ✅
</pre>
</details>
<p>

From Circle CI [build 37](https://circleci.com/gh/vivlabs/coverage-github-reporter/37)  — generated by [coverage-github-reporter](https://github.com/vivlabs/coverage-github-reporter)

Commit:
more logging

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/38/artifacts/0//coverage/lcov-report/index.html): 67.49% 💛** 

<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/38/artifacts/0//coverage/lcov-report/index.html">src/         </a>    23.17% 💔
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/38/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/</a>   100.00% ✅
</pre>
</details>
<p>

From Circle CI [build 38](https://circleci.com/gh/vivlabs/coverage-github-reporter/38)  — generated by [coverage-github-reporter](https://github.com/vivlabs/coverage-github-reporter)

Commit:
fix logic bug

Commit:
show new folders

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/index.html): 67.45% 💛 +12.36% 😀** 
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/index.html">src/               </a>    22.95% 💔  -32.13% 😡
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/bot.js.html">  bot.js           </a>    21.79% 💔  +21.79% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/curl.js.html">  curl.js          </a>    50.00% 💔  +50.00% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/github-comment.js.html">  github-comment.js</a>    33.38% 💔  +33.38% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/      </a>   100.00% ✅
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/index.html">src/         </a>    22.95% 💔
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/40/artifacts/0//coverage/lcov-report/coverage/index.html">src/coverage/</a>   100.00% ✅
</pre>
</details>
<p>

From Circle CI [build 40](https://circleci.com/gh/vivlabs/coverage-github-reporter/40) (compared to [build 23](https://circleci.com/gh/vivlabs/coverage-github-reporter/23) of `master` branch) — generated by [coverage-github-reporter](https://github.com/vivlabs/coverage-github-reporter)

Commit:
tweak github comment format

Commit:
more shared folder calculation

viv-9000 commented:
**[Code Coverage](https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/index.html): 68.32% 💛 +13.23% 😀** 
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/index.html">src/               </a>    22.95% 💔  -32.13% 😡
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/bot.js.html">  bot.js           </a>    21.79% 💔  +21.79% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/curl.js.html">  curl.js          </a>    50.00% 💔  +50.00% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/github-comment.js.html">  github-comment.js</a>    33.38% 💔  +33.38% 😀
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/coverage/index.html">src/coverage/      </a>   100.00% ✅
</pre>
<details>
<summary>Folder Coverage</summary>
<pre>
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/index.html">src/         </a>    22.95% 💔
<a href="https://circleci.com/gh/vivlabs/coverage-github-reporter/42/artifacts/0//coverage/lcov-report/src/coverage/index.html">src/coverage/</a>   100.00% ✅
</pre>
</details>
<p>

Circle CI [build 42](https://circleci.com/gh/vivlabs/coverage-github-reporter/42) (compared to [build 23](https://circleci.com/gh/vivlabs/coverage-github-reporter/23) of `master` branch) — [coverage-github-reporter](https://github.com/vivlabs/coverage-github-reporter)

Commit:
more comment tweaks

Commit:
changelog updates

Commit:
v1.1.0-alpha1

mufucaw commented on src/coverage/format.js:
```diff
@@ -0,0 +1,145 @@
+
+const ALL_FILES_PATH = '*'
+
+const ERROR_THRESHOLD = 50
```
Do you think it would be useful to stat (and read if it exists) a .cgr/.coverage-github-reporter/etc file containing overrides for these values? Maybe this is a 2.0 feature? The consts would be defaults if the config doesn't exist in that case.

mufucaw commented on src/coverage/format.js:
```diff
@@ -0,0 +1,145 @@
+
+const getEmoji = (percent) => {
+  if (percent === 0) {
+    return '❌'
```
I suppose these could be customizable too in a .settings file in home, if that feature is added.
